protocols/rendezvous/src/server.rs

```diff
@@ -483,7 +501,27 @@ impl Registrations {
     }
 
     fn poll(&mut self, cx: &mut Context<'_>) -> Poll<ExpiredRegistration> {
-        let expired_registration = ready!(self.next_expiry.poll_next_unpin(cx)).expect(
+        if let Poll::Ready(Some(expired_registration)) = self.next_expiry.poll_next_unpin(cx) {
+            // clean up our cookies
+            self.cookies.retain(|_, registrations| {
+                registrations.remove(&expired_registration);
+
+                // retain all cookies where there are still registrations left
+                !registrations.is_empty()
+            });
+
+            self.registrations_for_peer
+                .remove_by_right(&expired_registration);
+
+            return match self.registrations.remove(&expired_registration) {
+                None => self.poll(cx),
+                Some(registration) => Poll::Ready(ExpiredRegistration(registration)),
+            };
+        }
+
+        Poll::Pending
+
+        /*let expired_registration = ready!(self.next_expiry.poll_next_unpin(cx)).expect(
```

Comment:
Did you mean to leave this commented out?

If we only have a single inner state machine to poll, I'd prefer if we stick to the `ready!` macro.